src/main/java/mods/railcraft/common/blocks/tracks/behaivor/CollisionHandler.java

```diff
@@ -48,11 +50,24 @@ public void onEntityCollidedWithBlock(World world, BlockPos pos, IBlockState sta
             if (node.getChargeGraph().getCharge() > 2000) {
                 boolean shock = true;
                 ItemStack overalls = getOveralls(entity);
-                if (overalls != null) {
+                ItemStack boots = getRubberBoots(entity);
+                if ((overalls != null) && (boots != null)) {
+                    shock = false;
+                    if (MiscTools.RANDOM.nextInt(300) == 0)
+                        entity.setItemStackToSlot(EntityEquipmentSlot.LEGS, InvTools.damageItem(overalls, 1));
+                    if (MiscTools.RANDOM.nextInt(300) == 150)
```

Comment:
Is this supposed to be A or B? You may be missing an else, otherwise this entire block seems pointless.